### ISSUE TITLE
TSX fails when path has '-'

### ISSUE THREAD

**Issue by Noriller** [privatenumber/tsx#473]
### Acknowledgements

- [X] I searched existing issues before opening this one to avoid duplicates
- [X] I understand this is not a place for seek help, but to report a bug
- [X] I understand that the bug must be proven first with a minimal reproduction
- [X] I will be polite, respectful, and considerate of people's time and effort

### Minimal reproduction URL

https://stackblitz.com/edit/node-pjfw48

### Version

v4.7.0

### Node.js version

v18.18.0

### Package manager

npm

### Operating system

Linux

### Problem & Expected behavior

- What happened?
When path has `'-'` TSX fails with `Invalid define value (must be an entity name or valid JSON syntax)`
- What did you expect instead?
Escape the path and work.

I also have this link for other stackblitz reproducing problem:
https://stackblitz.com/edit/node-jz6fc4

It also happens on windows.

I know this is a weird problem, I was helping to debug an error for someone on windows that has in their user that emoji like `User '-'` and they were having this problem, but the problem also affects linux.

- Full stacktrace
```
~/projects/node-pjfw48
❯ npm start

> start
> tsx "test '-'/index.ts"

(node:39) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
Error: Transform failed with 1 error:
error: Invalid define value (must be an entity name or valid JSON syntax): 'file:///home/projects/node-pjfw48/test%20'-'/index.ts'
    at failureErrorWithLog (/home/projects/node-pjfw48/node_modules/.pnpm/esbuild@0.15.11/node_modules/esbuild/lib/main.js:1560:15)
    at eval (/home/projects/node-pjfw48/node_modules/.pnpm/esbuild@0.15.11/node_modules/esbuild/lib/main.js:799:29)
    at responseCallbacks.<computed> (/home/projects/node-pjfw48/node_modules/.pnpm/esbuild@0.15.11/node_modules/esbuild/lib/main.js:665:9)
    at handleIncomingPacket (/home/projects/node-pjfw48/node_modules/.pnpm/esbuild@0.15.11/node_modules/esbuild/lib/main.js:720:9)
    at Socket.readFromStdout (/home/projects/node-pjfw48/node_modules/.pnpm/esbuild@0.15.11/node_modules/esbuild/lib/main.js:641:7)
    at EventEmitter.emit (node:events:42:9202)
    at addChunk (node:internal/streams/readable:105:4109)
    at readableAddChunk (node:internal/streams/readable:105:3808)
    at Readable.push (node:internal/streams/readable:105:2679)
    at _0x77f04e.onStreamRead (node:internal/stream_base_commons:158:2341) {
  errors: [
    {
      detail: undefined,
      id: '',
      location: null,
      notes: [],
      pluginName: '',
      text: "Invalid define value (must be an entity name or valid JSON syntax): 'file:///home/projects/node-pjfw48/test%20'-'/index.ts'"
    }
  ],
  warnings: []
}
```

### Contributions

- [ ] I plan to open a pull request for this issue
- [ ] I plan to make a financial contribution to this project

**Comment by privatenumber** [comment]
Thanks for the report!

I think this needs to be `JSON.stringify`: https://github.com/privatenumber/tsx/blob/985bbb8cff1f750ad02e299874e542b6f63495ef/src/utils/transform/index.ts#L49

Would you mind submitting a PR?

**Comment by privatenumber** [comment]
:tada: This issue has been resolved in v4.7.1

If you appreciate this project, please consider [supporting this project by sponsoring](https://github.com/sponsors/privatenumber) :heart: :pray:
